src/internal/util/isPromise.ts

```diff
@@ -3,6 +3,6 @@
  * @see {@link https://www.ecma-international.org/ecma-262/6.0/#sec-promise-objects}
  * @param value the object to test
  */
-export function isPromise(value: any): value is PromiseLike<any> {
-  return !!value && typeof (<any>value).subscribe !== 'function' && typeof (value as any).then === 'function';
+export function isPromise<T = any>(value: any): value is PromiseLike<T> {
+  return !!value && typeof value.subscribe !== 'function' && typeof value.then === 'function';
```

Comment:
IMO, this change is unnecessary and is incorrect. Essentially, it uses a type parameter as a type assertion and I think that should be avoid. The current types accurately describe what this user-defined type-guard function does.

Reply:
- this situation cannot be deduced correctly
![image](https://user-images.githubusercontent.com/25052421/69912719-32396d00-1468-11ea-9132-c0624d950c49.png)

- I hope so
![image](https://user-images.githubusercontent.com/25052421/69912728-4da47800-1468-11ea-83b6-64519c94179e.png)

Reply:
If `isPromise` is used in that way in this package, the usage should be changed - `Promise<string> | any` is equivalent to `any`, so there is no inference that can be drawn from that type.

And `isPromise` is an internal function, so it should not use used outside of this package - it is not exported, AFAICT.

And the the snippet you have above illustrates my point. It isn't inferring anything; it's a type parameter acting as a type assertion. IMO, it is better to be explicit with a type assertion.

Reply:
Okay, I didn't notice `isPromise` is an internal function.